examples/3d/load_gltf_animation.rs

```diff
@@ -0,0 +1,210 @@
+) {
+    // Insert a resource with the current scene information
+    commands.insert_resource(CurrentScene {
+        // Its instance id, to be able to check that it's loaded
+        instance_id: scene_spawner.spawn(asset_server.load(&format!("{}#Scene0", ANIMATIONS[0].0))),
+        // The handle to the first animation
+        animation: asset_server.load(&format!("{}#Animation0", ANIMATIONS[0].0)),
+        // The animation speed modifier
+        speed: ANIMATIONS[0].2,
+    });
+
+    // Add a camera
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: ANIMATIONS[0].1,
+        ..Default::default()
+    });
+
+    // Add a directional light
+    commands.spawn_bundle(DirectionalLightBundle::default());
+}
+
+// Switch the scene to the next one every 10 seconds
+fn switch_scene(
+    mut commands: Commands,
+    scene_root: Query<Entity, (Without<Camera>, Without<DirectionalLight>, Without<Parent>)>,
+    mut camera: Query<&mut Transform, With<Camera>>,
+    mut current: Local<usize>,
+    mut current_scene: ResMut<CurrentScene>,
+    asset_server: Res<AssetServer>,
+    mut scene_spawner: ResMut<SceneSpawner>,
+) {
+    *current = (*current + 1) % ANIMATIONS.len();
+
+    // Despawn the existing scene, then start loading the next one
+    commands.entity(scene_root.single()).despawn_recursive();
+    current_scene.instance_id =
+        scene_spawner.spawn(asset_server.load(&format!("{}#Scene0", ANIMATIONS[*current].0)));
+    current_scene.animation = asset_server.load(&format!("{}#Animation0", ANIMATIONS[*current].0));
+    current_scene.speed = ANIMATIONS[*current].2;
+
+    // Update the camera position
+    *camera.single_mut() = ANIMATIONS[*current].1;
+
+    // Reset the current animation
+    commands.remove_resource::<CurrentAnimation>();
+}
+
+// Setup the scene for animation once it is loaded, by adding the animation to a resource with
+// the start time.
+fn setup_scene_once_loaded(
+    mut commands: Commands,
+    scene_spawner: Res<SceneSpawner>,
+    current_scene: Res<CurrentScene>,
+    time: Res<Time>,
+    mut done: Local<bool>,
+    animations: Res<Assets<GltfAnimation>>,
+) {
+    // If the current scene resource has changed, start waiting for it to be loaded
+    if current_scene.is_changed() {
+        *done = false;
+    }
+    // Once the scene and the animation are loaded, start the animation
+    if !*done && scene_spawner.instance_is_ready(current_scene.instance_id) {
+        if let Some(animation) = animations.get(&current_scene.animation) {
+            *done = true;
+            commands.insert_resource(CurrentAnimation {
+                start_time: time.seconds_since_startup(),
+                animation: animation.clone(),
+            })
+        }
+    }
+}
+
+// This animation driver is not made to work in the general case. It will work with only one
+// animation per scene, and will ignore the specified interpolation method to only do linear.
+fn gltf_animation_driver(
```

Comment:
It's probably worth pointing out here that this isn't completely implemented per the spec. It in fact only animates transforms for GLTF nodes and will not animated bones or morphs which are also apart of the spec. I don't see any changes to the GLTF loader to load the skin information as well. I think this is a great PR though and I don't think those things should hold back it! We should just make it clear what's not currently supported here. 🤔

Reply:
Skin will be added in #2359, and morphs in #3722. This PR should work directly for skin, but morph will require additional work to be animated, currently they will log a warning when loading a file with morph animation